### PR TITLE
Use SysCTypes and SysBasic by default in c2chapel

### PR DESCRIPTION
**Pull request by bmcdonald3**
Pretty much all c2chapel programs use SysCTypes and
SysBasic for all C types and C type aliases, but
it previously wasn't a default `use`, only CPtr was,
so this PR adds that to all c2chapel output and also
updates all existing tests to reflect that behavior.

Motivated by: https://github.com/Cray/chapel-private/issues/2294
Related to: https://github.com/Cray/chapel-private/issues/2363

### CODE REVIEW

File: tools/c2chapel/c2chapel.py [hunk 1]
```diff
@@ -61,8 +61,8 @@
 ## be registered here.
 c2chapel = {}
 
-c2chapel["double"] = "c_double"
-c2chapel["float"]  = "c_float"
+c2chapel["double"] = "real(64)"
+c2chapel["float"]  = "real(32)"
```
Comment (ronawho) [inline tools/c2chapel/c2chapel.py:65]: What's the motivation for this change? Seems like we'd want to use the c_ type.

Comment (bmcdonald3) [reply tools/c2chapel/c2chapel.py:65]: The `c_double` was causing an error saying it didn't exist and I found this on the [docs](https://chapel-lang.org/docs/1.16/technotes/extern.html):
```
For consistency, the following type aliases are also provided even though their sizes can't vary in C 
(thereby permitting the equivalent Chapel types to always be usable):

c_float  // (a real(32) in Chapel)
c_double // (a real(64) in Chapel)
```
so I changed those to just be their equivalents and didn't have any problems, so I thought it should be kept. Do you think I should move it back? Would just need to figure out why that c_double and c_float weren't working.

Comment (bmcdonald3) [reply tools/c2chapel/c2chapel.py:65]: I guess, for more background, I figured out they all need SysCTypes when I tried compiling the generated files and none of them worked without that and then the other thing that was preventing them from compiling was the c_double and c_float, so when I put this change in it resolved it and they all compile now. I'm not sure if I'm missing an include or something though

Comment (ronawho) [reply tools/c2chapel/c2chapel.py:65]: Right `c_double` should be an alias for `real(64)`, but since we're using c types it makes sense to me to use the `c_double` alias.

I'm not sure I follow your last 2 comments completely though. What didn't compile before?

Comment (bmcdonald3) [reply tools/c2chapel/c2chapel.py:65]: Trying to run this program:
```chapel
use SysCTypes;

var a:c_double;
```

I get this error:
`tools/c2chapel/test/asd.chpl:3: error: 'c_double' undeclared (first use this function)`

Comment (ronawho) [reply tools/c2chapel/c2chapel.py:65]: I see -- it looks like `c_double` is defined in SysBasic.chpl. I'm not sure why, but if you import that it should work.

Comment (ronawho) [reply tools/c2chapel/c2chapel.py:65]: From the header it says:
> This module provides basic types when those types are not
> platform-dependent. The module :mod:`SysCTypes` contains the
> platform-dependent declarations for C compatibility.

And since like your comment above says c_double isn't system dependent we must stuff it here instead of SysCTypes.

Comment (bmcdonald3) [reply tools/c2chapel/c2chapel.py]: Alright, added

Comment (ronawho) [reply tools/c2chapel/c2chapel.py]: Can you update PR title/description to reflect this change?

Comment (bmcdonald3) [reply tools/c2chapel/c2chapel.py]: Yup, done

**Review by ronawho — APPROVED** [review]
